tests/launch-fastify.test.js

```diff
@@ -172,62 +175,182 @@ test('Log level inheriting system with defaults checking data are properly strea
+  t.test('fields values', async assert => {
+    assert.plan(13)
+    const stream = split(JSON.parse)
+
+    stream.once('data', () => {
+      stream.once('data', line => {
+        assert.equal(line.reqId, '34')
+        assert.equal(line.level, 10)
+        assert.notOk(line.req)
+        assert.strictSame(line.http, {
           request: {
             method: 'GET',
             userAgent: { original: 'lightMyRequest' },
           },
-          response: {
-            statusCode: 200,
-            body: { bytes: '13' },
-          },
         })
-        assert.strictSame(secondLine.url, { path: '/' })
-        assert.strictSame(secondLine.host, { hostname: 'testHost', forwardedHost: 'testForwardedHost', ip: 'testIp' })
+        assert.strictSame(line.url, { path: '/' })
+        assert.strictSame(line.host, { hostname: 'testHost', forwardedHostame: 'testForwardedHost', ip: 'testIp' })
+
+        stream.once('data', secondLine => {
+          assert.equal(line.reqId, '34')
+          assert.equal(secondLine.level, 30)
+          assert.notOk(secondLine.res)
+          assert.ok(secondLine.responseTime)
+          assert.strictSame(secondLine.http, {
+            request: {
+              method: 'GET',
+              userAgent: { original: 'lightMyRequest' },
+            },
+            response: {
+              statusCode: 200,
+              body: { bytes: 13 },
+            },
+          })
+          assert.strictSame(secondLine.url, { path: '/' })
+          assert.strictSame(secondLine.host, { hostname: 'testHost', forwardedHost: 'testForwardedHost', ip: 'testIp' })
 
-        assert.end()
+          assert.end()
+        })
       })
     })
+
+    const fastifyInstance = await launch('./tests/modules/correct-module', {
+      logLevel: 'trace',
+      stream,
+    })
+    await fastifyInstance.inject({
+      method: 'GET',
+      url: '/',
+      headers: {
+        'x-forwarded-for': 'testIp',
+        'host': 'testHost:3000',
+        'x-forwarded-host': 'testForwardedHost',
+        'x-request-id': '34',
+      },
+    })
+
+    await fastifyInstance.close()
   })
 
-  const fastifyInstance = await launch('./tests/modules/correct-module', {
-    logLevel: 'trace',
-    stream,
+  t.test('matches schema', async assert => {
+    const ajv = new Ajv()
+
+    assert.plan(2)
+    const stream = split(JSON.parse)
+
+    const validator = ajv.compile(logSchema)
+
+    stream.once('data', () => {
+      stream.once('data', incomingRequest => {
+        const isValid = validator(incomingRequest)
+        if (validator.errors) {
+          // eslint-disable-next-line no-console
+          console.log('Validation errors', validator.errors)
+        }
+        assert.ok(isValid)
+
+        stream.once('data', requestCompleted => {
+          const isValid = validator(requestCompleted)
+          if (validator.errors) {
+            // eslint-disable-next-line no-console
+            console.log('Validation errors', validator.errors)
```

Comment:
We could use the t.fail here, or making the assertion on validation errors and set errors as message of the check instead of the console.log

Reply:
I've set a message and passing errors as extra